lib/content-services/src/lib/content-metadata/components/content-metadata/content-metadata.component.spec.ts

```diff
@@ -379,14 +391,180 @@ describe('ContentMetadataComponent', () => {
+    
+            expect(component.cancelChanges).toHaveBeenCalledWith(event);
+            expect(component.editable).toBe(false);
+        });
+    
+        it('should cancel tags changes and toggle editableTags flag', () => {
+            const event = new Event('click');
+            const buttonType = 'tags';
+            component.editableTags = true;
+    
+            spyOn(component, 'cancelChanges');
+            component.cancelEditChanges(buttonType, event);
+    
+            expect(component.cancelChanges).toHaveBeenCalledWith(event);
+            expect(component.editableTags).toBe(false);
+        });
+    
+        it('should cancel categories changes and toggle editableCategories flag', () => {
+            const event = new Event('click');
+            const buttonType = 'categories';
+            component.editableCategories = true;
+    
+            spyOn(component, 'cancelChanges');
+            component.cancelEditChanges(buttonType, event);
+    
+            expect(component.cancelChanges).toHaveBeenCalledWith(event);
+            expect(component.editableCategories).toBe(false);
+        });
+    });
+    
+
+    describe('editing', () => {
+        it('should toggle categories edit and set categoriesPanelState accordingly', () => {
+            const event = new Event('click');
+            spyOn(event, 'stopPropagation');
+            component.editableCategories = false;
+            component.categoriesPanelState = false;
+            component.toggleCategoriesEdit(event);
+            expect(event.stopPropagation).toHaveBeenCalled();
+            expect(component.editableCategories).toBe(true);
+            expect(component.categoriesPanelState).toBe(true);
+            component.toggleCategoriesEdit(event);
+            expect(component.editableCategories).toBe(false);
+            expect(component.categoriesPanelState).toBe(false);
+        });
+
+        it('should toggle group edit and expand the panel if editable', () => {
+            const event = new Event('click');
+            spyOn(event, 'stopPropagation');
+            const group = { editable: false, expanded: false };
+            component.toggleEdit(group, event);
+            expect(event.stopPropagation).toHaveBeenCalled();
+            expect(group.editable).toBe(true);
+            expect(group.expanded).toBe(true);
+        });
+        
+        it('should toggle group edit but not expand the panel if not editable', () => {
+            const event = new Event('click');
+            spyOn(event, 'stopPropagation');
+            const group = { editable: true, expanded: true };
+            component.toggleEdit(group, event);
+            expect(event.stopPropagation).toHaveBeenCalled();
+            expect(group.editable).toBe(false);
+            expect(group.expanded).toBe(true);
+        });
+        
+
+        it('should toggle general info edit and set generalInfoPanelState accordingly', () => {
```

Comment:
I don't understand, you set generalInfoPanelState to true at the beginning and test if that was set to true... You set it to true at the beginning so how do you know if code sets it to true too?